util/Pos/Util/Log/Scribes.hs

```diff
@@ -0,0 +1,134 @@
+-- | provide backends for `katip`
+module Pos.Util.Log.Scribes
+    ( mkStdoutScribe
+    , mkStderrScribe
+    , mkDevNullScribe
+    , mkFileScribe
+    , mkJsonFileScribe
+    ) where
+
+import           Universum hiding (fromString)
+
+import           Control.Exception.Safe (Exception (..), catchIO)
+
+import           Data.Aeson.Text (encodeToLazyText)
+import           Data.Text.Lazy.Builder
+import qualified Data.Text.Lazy.IO as T
+import           Katip.Core
+import           Katip.Format.Time (formatAsIso8601)
+import           Katip.Scribes.Handle (brackets, getKeys)
+import           System.FilePath ((</>))
+import           System.IO (BufferMode (LineBuffering), Handle,
+                     IOMode (WriteMode), hFlush, hSetBuffering, stderr, stdout)
+import           System.IO.Unsafe (unsafePerformIO)
+
+import qualified Pos.Util.Log.Internal as Internal
+
+
+-------------------------------------------------------------------------------
+
+
+
+-- | global lock for file Scribes
+{-# NOINLINE lock #-}
+lock :: MVar ()
+lock = unsafePerformIO $ newMVar ()
+
+prtoutException :: Exception e => FilePath -> e -> IO ()
+prtoutException fp e = do
+    putStrLn $ "error while opening log @ " ++ fp
+    putStrLn $ "exception: " ++ displayException e
+
+-- | create a katip scribe for logging to a file (JSON)
+mkJsonFileScribe :: FilePath -> FilePath -> Severity -> Verbosity -> IO Scribe
+mkJsonFileScribe bp fp s v = do
+    h <- catchIO (openFile (bp </> fp) WriteMode) $
+            \e -> do
+                prtoutException (bp </> fp) e
+                return stdout    -- fallback to standard output in case of exception
+    hSetBuffering h LineBuffering
+    _lock <- newMVar ()
```

Comment:
Why does `_lock` have a leading underscore? Normally that is used to tell GHC to ignore the fact that the variable is not used elsewhere, but in this case its used on line 54 below.

Reply:
OK, bad choice.